### ISSUE TITLE
What is the best way to draw train/valid loss chart?

### ISSUE THREAD

**Issue by ZetilenZoe** [AlexeyAB/darknet#58]
In your detailed tutorial on training for own dataset, you mention about early stopping in order to avoid over-fitting. Is there any easy way to obtain the train/valid loss chart? The only way that I can imagine is getting the weights from several iterations and computing loss for them. But this is a tiresome approach.

**Comment by AlexeyAB** [comment]
@ZetilenZoe 

There is no any way to do this. Because you should do valid for each images of validation-dataset to get **average IoU**.

To simplify, I suggest looking at the current indicator **avg loss**, that much less accurate, but visible immediately. And only when decreasing of avg loss stoped, then you should do valid.

**Comment by phongnhhn92** [comment]
I have the same problem but I want to draw it after training.
Is there any way, I can see all the log file after training finished ?

**Comment by sivagnanamn** [comment]
Everyone who has used other Deep learning frameworks like TensorFlow, Keras etc are used to checking train & validation error during training.

> To simplify, I suggest looking at the current indicator avg loss, that much less accurate, but visible immediately. And only when decreasing of avg loss stoped, then you should do valid.

@AlexeyAB  It would be helpful if you could elaborate the reason behind the above statement. Thank you.

**Comment by AlexeyAB** [comment]
@sivagnanamn It's just not implemented yet in the Darknet. So if you can implement it in the source code - yes it will be convenient.
